src/seedu/addressbook/data/person/addresscomponents/Unit.java

```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.person.addresscomponents;
+
+/** @@author kaisertan
+ *
+ * Represents the Unit number in an Address
+ *
+ */
+
+public class Unit {
+
+    private String value;
+    private boolean hasUnitNumber = true;
+    private final String NIL_INDICATOR = "0";
+
+
+    public Unit(String unitNumber){
+
+        this.value = unitNumber;
+    }
+
+    /**
+     *
+     * Returns the unit number in String if unit number is valid
```

Comment:
Header comment mentions validity check, but there doesn't seem to be such a check here.